crates/proof-of-sql/src/sql/postprocessing/order_by_exprs.rs

```diff
@@ -0,0 +1,73 @@
+use super::{PostprocessingError, PostprocessingResult, PostprocessingStep};
+use crate::base::{
+    database::{compare_indexes_by_owned_columns_with_direction, OwnedColumn, OwnedTable},
+    math::permutation::Permutation,
+    scalar::Scalar,
+};
+use proof_of_sql_parser::intermediate_ast::{OrderBy, OrderByDirection};
+use rayon::prelude::ParallelSliceMut;
+use serde::{Deserialize, Serialize};
+
+/// A node representing a list of `OrderBy` expressions.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct OrderByExprs<S: Scalar> {
+    by_exprs: Vec<OrderBy>,
+    _phantom: core::marker::PhantomData<S>,
+}
+
+impl<S: Scalar> OrderByExprs<S> {
+    /// Create a new `OrderByExprs` node.
+    pub fn new(by_exprs: Vec<OrderBy>) -> Self {
+        Self {
+            by_exprs,
+            _phantom: core::marker::PhantomData,
+        }
+    }
+}
+
+impl<S: Scalar> PostprocessingStep<S> for OrderByExprs<S> {
+    /// Apply the slice transformation to the given `OwnedTable`.
+    fn apply(&self, owned_table: OwnedTable<S>) -> PostprocessingResult<OwnedTable<S>> {
+        let mut indexes = (0..owned_table.num_rows()).collect::<Vec<_>>();
+        // Evaluate the columns by which we order
+        // Once we allow OrderBy for general aggregation-free expressions here we will need to call eval()
+        let order_by_pairs: Vec<(OwnedColumn<S>, OrderByDirection)> = self
+            .by_exprs
+            .iter()
+            .map(
+                |order_by| -> PostprocessingResult<(OwnedColumn<S>, OrderByDirection)> {
+                    Ok((
+                        owned_table
+                            .inner_table()
+                            .get(&order_by.expr)
+                            .ok_or(PostprocessingError::ColumnNotFound(
+                                order_by.expr.to_string(),
+                            ))?
+                            .clone(),
+                        order_by.direction,
+                    ))
+                },
+            )
+            .collect::<PostprocessingResult<Vec<(OwnedColumn<S>, OrderByDirection)>>>()?;
+        // Define the ordering
+        indexes.par_sort_unstable_by(|&a, &b| {
+            compare_indexes_by_owned_columns_with_direction(&order_by_pairs, a, b)
+        });
+        let permutation = Permutation::try_new(indexes)
+            .expect("Permutating any permutation results in another valid permutation");
```

Comment:
I would think that an `unchecked_new` should be used here since you are expecting away the `try`. Also, in this case the `try_new` is doing non-trivial work.
```suggestion
        let permutation = Permutation::unchecked_new(indexes);
```